MidpointFindingKarel.java

```diff
@@ -15,5 +15,38 @@
 public class MidpointFindingKarel extends SuperKarel {
 
 	// You fill in this part
-
+	public void run(){
+		
+		int count = 1;
+		while(true)
+		{
+			if(frontIsClear())
+			{
+				move();
+				count++;
+			}
+			else
+			{
+				turnAround();
+				break;
+			}
+		}
+		
+		for(int i=0;i<count/2;i++)
+		{
+			move();
+			
+		}
+			if(count%2==0)
```

Comment:
This is wrong, please go and check the problem statement. This is not what problem statement says.